### PR TITLE
Fix right alignment in RichTextLabel

### LINKED ISSUE #54764: [3.x] RichTextLabel issue starting 3.4
### Godot version

3.4.stable

### System information

Win10, Linux

### Issue description

After the recent upgrade from 3.3.stable to 3.4.stable the RichTextLabel is behaving weirdly when using conditional positioning in the bbcode.
Example project attached. Just play with the text inside of the bbcode_text block - it looks like it is calculating too little space and, as soon as it is just "1 word" it just breaks into a "new line" forcing the right-aligned text to be left aligned. This happens exclusively to right aligned text.

This is a task open by proxy for Tosek on the German Discord Community Server - I will communications between the parties incase there is a language barrier in place.

To clarify: This issue also happens for me aswell, I tried it on the latest 3.4.stable version on ArchLinux and on Windows.

### Steps to reproduce

Have big font size, have text in RichTextEditor. Have a "[right]text with spaces[/right]" bb-block. Expect proper text alignment, get broken text alignment and weird new line breaks.

### Minimal reproduction project

[RichTextLabelBug.zip](https://github.com/godotengine/godot/files/7498313/RichTextLabelBug.zip)

This contains the "arial" font aswell, incase anyone is concerned about the license.
I tried it with other fonts aswell (Comme Sans), same effect.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #54764

### CODE REVIEW

**Comment by pycbouh** [comment]
Does it fix https://github.com/godotengine/godot/issues/54551?

**Comment by bruvzg** [comment]
> Does it fix #54551?

No, it's not fixing it, probably some other `int` → `float` are still missing.

**Review by akien-mga — APPROVED** [review]

File: scene/gui/rich_text_label.cpp [hunk 1]
```diff
@@ -222,7 +222,7 @@ int RichTextLabel::_process_line(ItemFrame *p_frame, const Vector2 &p_ofs, int &
 			if (align != ALIGN_FILL)                                                                                                                                        \
 				wofs += line_ofs;                                                                                                                                           \
 		} else {                                                                                                                                                            \
-			int used = wofs - margin;                                                                                                                                       \
+			float used = wofs - margin;                                                                                                                                       \
```
Comment (akien-mga) [inline scene/gui/rich_text_label.cpp:225]: Trailing backslash needs to be aligned.

**Comment by KoBeWi** [comment]
I checked #54551 and it's not obvious where to look for the fix. This PR makes the issue less severe though.

**Comment by bruvzg** [comment]
Here's a fix for the #54551 - #55440.

**Review by bruvzg — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.1.

**Pull request merged**